### PR TITLE
matter: samples: Integrated nRF7002 targets with sysbuild

### PR DESCRIPTION
**Pull request by markaj-nordic**
- added support for nRF7002 DK to the sysbuild scripts
- adapted Matter samples configuration to support nRF7002 DK builds with sysbuild

### CODE REVIEW

**Review by nordicjm — APPROVED** [review]

File: samples/matter/light_switch/Kconfig.sysbuild
Comment (doublemis1) [inline samples/matter/light_switch/Kconfig.sysbuild:10]: Question: if we build the nrf5340 with nrf7002ek we should expect to have multiprotocol rpmsg?

Comment (markaj-nordic) [reply samples/matter/light_switch/Kconfig.sysbuild:10]: Good point, initially I though the nrf7002ek was out of scope, but now I guess I can be handled. I will open the next PR for that. The same will apply to the Matter weather station and nrf7002eb, but this needs to wait until we have OTA support for sysbuild builds, because applications do not support non-DFU configuration.

Comment (markaj-nordic) [reply samples/matter/light_switch/Kconfig.sysbuild]: Ok, I fixed it here.

File: sysbuild/Kconfig.netcore
Comment (kkasperczyk-no) [inline sysbuild/Kconfig.netcore:43]: Could we follow the convention used for the other configs e.g NETCORE_802154_RPMSG -> NRF_DEFAULT_802154, NETCORE_HCI_RPMSG -> NRF_DEFAULT_HCI?

File: sysbuild/Kconfig.netcore
Comment (kkasperczyk-no) [inline sysbuild/Kconfig.netcore:140]: Is it correct to support nRF7002 with 802154 rpmsg image?

File: sysbuild/Kconfig.netcore
Comment (kkasperczyk-no) [inline sysbuild/Kconfig.netcore:148]: Ditto

File: sysbuild/Kconfig.netcore
Comment (kkasperczyk-no) [inline sysbuild/Kconfig.netcore:42]: This is minor, but I think it's not the best approach to copy+paste the same board names for every default option. Considering the fact we already have all of them listed in `config SUPPORT_NETCORE` and additionally every `NRF_DEFAULT_X` depends on some specific boards, wouldn't be enough to just add `if SUPPORT_NETCORE` before the `choice NETCORE`?

File: samples/matter/window_covering/Kconfig.sysbuild
Comment (kkasperczyk-no) [inline samples/matter/window_covering/Kconfig.sysbuild:14]: We do not support Wi-Fi for window covering.

File: samples/matter/light_bulb/Kconfig.sysbuild
Comment (nordicjm) [inline samples/matter/light_bulb/Kconfig.sysbuild:14]: I do not find it acceptable for sysbuild to be loading board specific shield files

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: samples/matter/light_bulb/Kconfig.sysbuild
Comment (markaj-nordic) [reply samples/matter/light_bulb/Kconfig.sysbuild:14]: Discussed f2f. Will be reverted.

File: sysbuild/Kconfig.netcore
Comment (markaj-nordic) [reply sysbuild/Kconfig.netcore:43]: @nordicjm are you fine with that? This would affect > 50 files that use this symbol.

File: sysbuild/Kconfig.netcore
Comment (markaj-nordic) [reply sysbuild/Kconfig.netcore:140]: @LuDuda? I don't see the reason why not. Having Wi-Fi and Thread without BLE sounds sane to me for some use cases.

File: samples/matter/window_covering/Kconfig.sysbuild
Comment (markaj-nordic) [reply samples/matter/window_covering/Kconfig.sysbuild:14]: This change will be reverted anyway.

File: sysbuild/Kconfig.netcore
Comment (nordicjm) [reply sysbuild/Kconfig.netcore:43]: No. The original idea was for bluetooth to signal bluetooth support and 802154 to signal 802.15.4 support, so that if both were selected then it would choose the multiprotocol image, but this is currently on hold, it might be revived in future.

Comment (markaj-nordic) [reply sysbuild/Kconfig.netcore:43]: But, the question was only about changing the name `NRF_DEFAULT_BLUETOOTH` -> `NRF_DEFAULT_HCI`. Nevermind, I will not change it.

File: sysbuild/Kconfig.netcore
Comment (markaj-nordic) [reply sysbuild/Kconfig.netcore:42]: I see. @nordicjm, @tejlmand can we do what Kamil suggested? Or maybe something like the following would work?

choice NETCORE
	prompt "Netcore image"
	default NETCORE_EMPTY if NRF_DEFAULT_EMPTY && SUPPORT_NETCORE
	default NETCORE_HCI_RPMSG if NRF_DEFAULT_BLUETOOTH && SUPPORT_NETCORE
	default NETCORE_802154_RPMSG if NRF_DEFAULT_802154 && SUPPORT_NETCORE
	default NETCORE_MULTIPROTOCOL_RPMSG if NRF_DEFAULT_MULTIPROTOCOL && SUPPORT_NETCORE
	default NETCORE_NONE
	depends on SUPPORT_NETCORE

File: sysbuild/Kconfig.netcore
Comment (LuDuda) [reply sysbuild/Kconfig.netcore:140]: This is correct configuration i.e. nRF7002 + 802154 rpmsg image, we do have a client that is going to use it. Though, normal case is hci rpmsg only.

Comment (LuDuda) [reply sysbuild/Kconfig.netcore:140]: So.. nRF7002 + 802154 would mean Wi-Fi + Thread - BLE. While i can see this configuration valid, we do not have any use-case at least in Matter for this.

**Comment by kkasperczyk-no** [comment]
@nordicjm do you have any more comments or can this PR get your approval?

**Review by nordicjm — APPROVED** [review]

**Pull request merged**
